core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala

```diff
@@ -93,8 +94,8 @@ class NettyBlockRpcServer(
           } else {
             val startAndEndId = fetchShuffleBlocks.reduceIds(index)
             if (startAndEndId.length != 2) {
-              throw new IllegalStateException(s"Invalid shuffle fetch request when batch mode " +
-                s"is enabled: $fetchShuffleBlocks")
+              throw SparkException.internalError(s"Invalid shuffle fetch request when batch mode " +
+                s"is enabled: $fetchShuffleBlocks", category = "INTERNAL_ERROR_NETWORK")
```

Comment:
Just `NETWORK`, I think
```suggestion
                s"is enabled: $fetchShuffleBlocks", category = "NETWORK")
```